odbc/src/api/errors.rs

```diff
@@ -1,49 +1,50 @@
-const VENDOR_IDENTIFIER: &str = "MongoDB";
+use constants::{HY024, HYC00, VENDOR_IDENTIFIER, _01S02};
+use thiserror::Error;
 
-// SQL states
-pub const HYC00: &str = "HYC00";
-pub const HY024: &str = "HY024";
-pub const _01S02: &str = "01S02";
-
-#[derive(Debug)]
+#[derive(Debug, Error)]
 pub enum ODBCError {
+    #[error("[{}][API] The feature {0} is not implemented", VENDOR_IDENTIFIER)]
     Unimplemented(&'static str),
+    #[error("[{}][API] Invalid Uri {0}", VENDOR_IDENTIFIER)]
+    InvalidUriFormat(String),
+    #[error("[{}][API] Invalid handle type, expected {0}", VENDOR_IDENTIFIER)]
+    InvalidHandleType(&'static str),
+    #[error("[{}][API] Invalid value for attribute {0}", VENDOR_IDENTIFIER)]
     InvalidAttrValue(&'static str),
+    #[error(
+        "[{}][API] Invalid value for attribute {0}, changed to {1}",
+        VENDOR_IDENTIFIER
+    )]
     OptionValueChanged(&'static str, &'static str),
+    #[error("[{}][Core] {0}", VENDOR_IDENTIFIER)]
+    CoreError(#[from] mongo_odbc_core::Error),
 }
 
+pub type Result<T> = std::result::Result<T, ODBCError>;
+
 impl ODBCError {
     pub fn get_sql_state(&self) -> &str {
         match self {
             ODBCError::Unimplemented(_) => HYC00,
-            ODBCError::InvalidAttrValue(_) => HY024,
+            // TODO: set the state based on the type of CoreError since they may differ
```

Comment:
This TODO seems addressed given the case on line 30. Is this still a TODO that is being tracked by a future ticket? If yes, should we note the ticket number as part of this TODO?

Reply:
TODO is stale.